### ISSUE TITLE
[a11y] donation banner has two main concerns

### ISSUE THREAD

**Issue by CharlesNepote** [openfoodfacts/openfoodfacts-server#10298]
The donation banner is displayed only with a mouse click: `<span id="hide-donate-banner" class="material-icons" onclick="DonationButton();">close</span>`.

The analyzer we use reports two main issues:
1.  The mouse event attribute doesn't have a keyboard equivalent: see https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR20
2.  The element is a forced user interface control without the proper role attribute: see https://www.w3.org/WAI/WCAG21/Techniques/failures/F59

For the first one, it should be possible to open it thanks to keyboard. The solution seems to only add a keyboard event:
```html
<span id="hide-donate-banner" class="material-icons" onclick="DonationButton();" onkeypress="DonationButton();" role="link">close</span>
```

The second one is more obscure...
